packages/core/datadog-sourcemaps.gradle

```
@@ -0,0 +1,131 @@
+
+        if (bundleTask == null) {
+            logger.info("Cannot find JS bundle task for variant=${targetName}.")
+            return
+        }
+        if (!bundleTask.enabled) {
+            logger.info("JS bundle task for variant=${targetName} is not enabled.")
+            return
+        }
+
+        def serviceName = getServiceName(variant)
+        logger.info("Service name used for the upload of variant=${targetName} is ${serviceName}.")
+
+        def bundleAssetName = reactConfig.bundleAssetName ?: "index.android.bundle"
+
+        def jsBundleDir = file("$buildDir/generated/assets/react/${targetPath}")
+        def jsBundleFile = file("$jsBundleDir/$bundleAssetName")
+
+        def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${targetPath}")
+        def jsOutputSourceMapFile = file("$jsSourceMapsDir/${bundleAssetName}.map")
+
+        def uploadTask = tasks.create("upload${targetName}Sourcemaps") {
+            group = "datadog"
+            description = "Uploads sourcemaps to Datadog."
+
+            def execCommand = [
+                    "yarn",
+                    "datadog-ci",
+                    "react-native",
+                    "upload",
+                    "--platform",
+                    "android",
+                    "--service",
+                    serviceName,
+                    "--bundle",
+                    jsBundleFile.absolutePath,
+                    "--sourcemap",
+                    jsOutputSourceMapFile.absolutePath,
+                    "--release-version",
+                    releaseVersion,
+                    "--build-version",
+                    buildVersion
+            ]
+
+            doFirst {
+                if (!jsBundleFile.exists()) {
+                    throw new GradleException("JS bundle file doesn't exist, aborting upload.")
+                }
+
+                if (!jsOutputSourceMapFile.exists()) {
+                    throw new GradleException("JS sourcemap file doesn't exist, aborting upload.")
+                }
+
+                runShellCommand(execCommand, reactRoot)
+
+            }
+        }
+
+        uploadTask.dependsOn bundleTask
+        bundleTask.finalizedBy uploadTask
+
+    }
+}
+
+private def runShellCommand(
+        List<String> command,
+        File workingDirectory
+) {
+    def outputStream = new ByteArrayOutputStream()
+    def errorStream = new ByteArrayOutputStream()
+
+    try {
+        def result = exec {
+            workingDir = workingDirectory
+            standardOutput = outputStream
+            errorOutput = errorStream
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                commandLine("cmd", "/c", *command)
+            } else {
+                commandLine(*command)
+            }
+        }
+
+        if (result.exitValue != 0) {
+            logger.error(errorStream.toString("UTF-8"))
+            result.rethrowFailure()
+        } else {
+            println outputStream.toString("UTF-8")
```

We can still use logger if we want to print it by default (without --info argument).

```suggestion
            logger.lifecycle(outputStream.toString("UTF-8"))
```